core/scheduler/service/job_lineage_service.go

```diff
@@ -0,0 +1,68 @@
+package service
+
+import (
+	"context"
+
+	"github.com/goto/salt/log"
+
+	"github.com/goto/optimus/core/scheduler"
+)
+
+// Contract that can be used by other callers to fetch job lineage information
+type JobLineageFetcher interface {
+	GetJobLineage(ctx context.Context, jobSchedules []*scheduler.JobSchedule) (map[scheduler.JobName]*scheduler.JobLineageSummary, error)
+}
+
+type LineageBuilder interface {
+	BuildLineage(context.Context, []*scheduler.JobSchedule, int) (map[*scheduler.JobSchedule]*scheduler.JobLineageSummary, error)
+}
+
+type JobLineageService struct {
+	l              log.Logger
+	lineageBuilder LineageBuilder
+}
+
+func (j *JobLineageService) GetJobExecutionSummary(ctx context.Context, jobSchedules []*scheduler.JobSchedule, numberOfUpstreamPerLevel int) ([]*scheduler.JobRunLineage, error) {
```

Comment:
@deryrahman , @ahmadnaufal can the service returning a map of JobSchedule & JobRunLineage will be better right?